tests/framework/installer/ceph_installer.go

```diff
@@ -314,31 +362,37 @@ func (h *CephInstaller) InstallRookOnK8sWithHostPathAndDevices(namespace, storeT
 		return false, err
 	}
 
-	if forceUseDevices {
-		logger.Infof("Forcing the use of devices")
-		useDevices = true
-	} else if useDevices {
-		// This check only looks at the local machine for devices. If you want to force using devices,
-		// set the forceUseDevices flag
-		useDevices = IsAdditionalDeviceAvailableOnCluster()
+	if usePVCs {
+		err = h.CreateK8sRookClusterWithPVCs(namespace, onamespace, storeType, useDevices,
+			cephv1.MonSpec{Count: mon.Count, AllowMultiplePerNode: mon.AllowMultiplePerNode}, false, rbdMirrorWorkers, h.CephVersion)
+	} else {
+		if forceUseDevices {
+			logger.Infof("Forcing the use of devices")
+			useDevices = true
+		} else if useDevices {
+			// This check only looks at the local machine for devices. If you want to force using devices,
+			// set the forceUseDevices flag
+			useDevices = IsAdditionalDeviceAvailableOnCluster()
+		}
+		// Create rook cluster
+		err = h.CreateK8sRookClusterWithHostPathAndDevices(namespace, onamespace, storeType,
+			useDevices, cephv1.MonSpec{Count: mon.Count, AllowMultiplePerNode: mon.AllowMultiplePerNode}, startWithAllNodes,
+			rbdMirrorWorkers, h.CephVersion)
 	}
-
-	// Create rook cluster
-	err = h.CreateK8sRookClusterWithHostPathAndDevices(namespace, onamespace, storeType,
-		useDevices, cephv1.MonSpec{Count: mon.Count, AllowMultiplePerNode: mon.AllowMultiplePerNode}, startWithAllNodes,
-		rbdMirrorWorkers, h.CephVersion)
 	if err != nil {
 		logger.Errorf("Rook cluster %s not installed, error -> %v", namespace, err)
 		return false, err
 	}
 
-	discovery, err := h.k8shelper.Clientset.AppsV1().DaemonSets(onamespace).Get("rook-discover", metav1.GetOptions{})
-	if startDiscovery {
-		assert.NoError(h.T(), err)
-		assert.NotNil(h.T(), discovery)
-	} else {
-		assert.Error(h.T(), err)
-		assert.True(h.T(), errors.IsNotFound(err))
+	if !usePVCs {
+		discovery, err := h.k8shelper.Clientset.AppsV1().DaemonSets(onamespace).Get("rook-discover", metav1.GetOptions{})
```

Comment:
Whether the discovery service is running is actually an independent setting from whether PVCs are being used. I'd suggest we not change the discovery daemonset in this test. We can just leave it running.

Reply:
cool removed it